docs/concepts/configuration/assign-pod-node.md

```diff
@@ -206,6 +206,94 @@ If defined but empty, it means "all namespaces."
 All `matchExpressions` associated with `requiredDuringSchedulingIgnoredDuringExecution` affinity and anti-affinity
 must be satisfied for the pod to schedule onto a node.
 
+#### More Practical Use-cases 
+
+Interpod Affinity and AnitAffinity can be even more useful when they are used with higher
+level collections such as ReplicaSets, Statefulsets, Deployments, etc.  One can easily configure that a set of workloads should
+be co-located in the same defined topology, eg., the same node. 
+
+##### Always co-located in the same node
+
+In a three node cluster, a web application has in-memory cache such as redis, we want the web-servers to be co-located with the cache as much as possible. 
+Here is the yaml snippet of a simple redis deployment with two replicas and selector label `app=store`
+
+```yaml
+apiVersion: apps/v1beta1 # for versions before 1.6.0 use extensions/v1beta1
+kind: Deployment
+metadata:
+  name: redis-cache
+spec:
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        app: store
+    spec:
+      containers:
+      - name: redis-server
+        image: redis:3.2-alpine
+```
+
+Below yaml snippet of the webserver deployment has `podAffinity` configured, this informs the scheduler that all its replicas are to be 
+co-located with pods that has selector label `app=store` 
+
+```yaml
+apiVersion: apps/v1beta1 # for versions before 1.6.0 use extensions/v1beta1
+kind: Deployment
+metadata:
+  name: web-server
+spec:
+  replicas: 4
+  template:
+    metadata:
+      labels:
+        app: web-store
+    spec:
+      affinity:
```

Comment:
Running multiple web-servers on a single machine rarely makes sense, especially if the intent is to handle traffic load. Well known web servers are multi-threaded and a single process can use multiple CPU cores on a machine. Besides, there is a potential of having port conflicts if you run multiple web server to serve one service point.
Scheduler tries to spread pods, but users shouldn't assume that pods of a single collection will never land on a single machine without proper anti-affinity rules.

All that said, I agree with you that for the sake of this example which is co-location we may want to keep the config simple. How about this:
1. Change the number of instances of redis and web servers to 3 instances, which is the same as the number of nodes.
2. Keep the rest of the config the same as it is now.
3. Add a note at the bottom of the example to point out that if spreading of web-servers or redis instances is desired, proper anti-affinity rules should be added and refer to the next example for anti-affinity.